mypy/checkexpr.py

```diff
@@ -655,13 +655,7 @@ def check_call(self, callee: Type, args: List[Expression],
                 callee = callee.copy_modified(ret_type=ret_type)
             return callee.ret_type, callee
         elif isinstance(callee, Overloaded):
-            # Type check arguments in empty context. They will be checked again
-            # later in a context derived from the signature; these types are
-            # only used to pick a signature variant.
-            self.msg.disable_errors()
-            arg_types = self.infer_arg_types_in_context(None, args)
-            self.msg.enable_errors()
-
+            arg_types = self.infer_arg_types_in_empty_context(args)
```

Comment:
At first I was sceptical about this, but after some thinking and playing it seems to me this is actually OK. Because under the hood all relevant possibilities (even the union math) go through `infer_overload_return_type`, that calls `check_call` with a `CallableType` item that, in turn, will trigger `infer_arg_types_in_context` using current matching overload item arg type as context. So it should be fine.